# 3.6. Lifetime Co-Expressed Transcript Analysis: AgingNet

Co-expression analysis consists of creating a network that represents the existing relationships between transcripts. In this network, called AgingNet, its nodes are represented by transcripts and their interrelationships by the similarity between their expression profiles throughout life. Furthermore, it is possible to identify groups of highly co-expressed transcripts, called co-expression modules, which may represent specific biological processes. Briefly, the similarity between all pairs of transcripts was calculated by the DTW-MIC metric on the ageCollapsed dataset, creating a similarity matrix. Then, the matrix was clustered using the HDBSCAN algorithm \(CAMPELLO; MOULAVI; SANDER, 2013\) to identify co-expression modules.

## 3.6.1. Similarity between Genes.

To determine the similarity of the expression profile between pairs of transcripts throughout life, a metric developed for time series, called DTW-MIC \(\(RICCADONNA et al., 2016\)\) was used. This metric is a combination of two others, Dynamic Time Warping \(DTW, \(ITAKURA, 1975\)\) and Maximal Information Coefficient \(MIC, \(ALBANESE et al., 2018\)\).

‌

DTW is a measure of the distance between two sequences, which takes into account temporal displacements. The DTW algorithm uses dynamic programming to find an optimal alignment between the two series, through a non-linear distortion of the time axes. The magnitude of this distortion is reflected in the dissimilarity value, and the similarity between the shapes of the curves has a greater impact on the DTW than the value of the point-to-point distance between the two time-series, as in the case of the Euclidean distance \( Figure 3.5\). To obtain the similarity measure \(DTWs\) from the dissimilarity distance \(DTWd\), the transformation DTWs = 1/\(1+ DTWd\) was used, where DTWd is the normalized distance of two-time series, calculated using the [dtw package](http://dtw.r-forge.r-project.org/).

![Figura 3.5. Algoritmo DTW  \(A\) ilustra como a dist&#xE2;ncia Euclidiana difere da DTW ao calcular a dist&#xE2;ncia entre duas s&#xE9;ries temporais com mesmo perfil de express&#xE3;o, por&#xE9;m deslocadas no tempo. \(B\) mostra a matriz de dist&#xE2;ncia gerada pela programa&#xE7;&#xE3;o din&#xE2;mica durante o processo de otimiza&#xE7;&#xE3;o para encontrar a melhor distor&#xE7;&#xE3;o entre as duas s&#xE9;ries temporais. Ilustra&#xE7;&#xE3;o retirada de \(CRIEL; TSIPORKOVA, 2006\).](https://lh6.googleusercontent.com/LXDBWhrMpbH6yyLsZdu0J3992H-zwmznIPNjZJcaoXR9PWxDpZavXc-Ic6DKjxvbvhS1sy-5LDCLDMFHQqjXuQcj7-7BzfXUHfW58UiZiH-kfUuU2RU5ODUe9j_S6_oBQQtOrPva=s0)

The MIC measure is a member of a family of statistics known as Maximal Information-based Nonparametric Exploration \(MINE\) and was developed to explore the relationships between two variables in multidimensional datasets. The two distinguishing features of the MIC are generality, being possible to capture relationships of different natures, and evenness, which is the property of penalizing similar noise levels in the same way, regardless of the nature of the relationship between the variables as represented in Figure 3.6.

![Figura 3.6. Propriedades de Detec&#xE7;&#xE3;o do Coeficiente de Informa&#xE7;&#xE3;o M&#xE1;xima \(MIC\). Imagem adaptada de \(RESHEF et al., 2011\), mostrando os tipos de rela&#xE7;&#xF5;es que podem ser encontradas entre duas vari&#xE1;veis e a influ&#xEA;ncia que o n&#xED;vel de ru&#xED;do tem sobre o MIC.](https://lh4.googleusercontent.com/5hcrX8xlPBmcUlXNb6AefMXOQ3eU_OgADqyREJ1qScNivz-bZJri5KFQsba4hdd0Ll-On7F7I3GWeJEd5KqS-rxpRrVVMzOZwFVCzHNWmIOrnukpKX0QnVwI-YpzA96UKvbOaOyg=s0)

As mentioned earlier, in this work a composition of DTW measures with MIC was used, called DTW-MIC \(RICCADONNA et al., 2016\) and defined as:

![](https://lh5.googleusercontent.com/3ckr9PMxLhWxLjpA9mD-rZHfe1SH1kEveeGMnNu9PiiEo0VzTudlMxBbR_Ut5QEdz9L3ZXwN6n9nNMLY-k_rGTUmDgcQFA0f326ez1IO09EuMpyi6iQgfJjdTE3T0cBPi83Y6PXk=s0)

This new definition combines contributions from both the DTW and the MIC, that is, it takes into account both temporal shifts and nonlinear correlations. As evidenced through simulations in \(RICCADONNA et al., 2016\), such a combination is more effective than Pearson's Correlation Coefficient, and also than DTW and MIC if considered separately.

Once the similarity has been calculated for each transcript pair, we obtain a similarity matrix where the number of columns and rows is equal to the number of transcripts and each cell of this matrix represents the similarity between two probes.  


3.6.2. Detection of Co-Expression Modules and Sub-Modules

The detection of co-expression modules was done by clustering the transcripts similarity matrix through a density-based algorithm, called HDBSCAN \("Density-Based Clustering of Applications with Noise \(DBSCAN\) and Related Algorithms" \[R package dbscan version 1.1 -2\],” \[nd\]\). This algorithm groups points in a given space that are very close to each other \(points with many neighbours\) and defines as outlier the isolated ones, located in low-density regions \(CAMPELLO; MOULAVI; SANDER, 2013\). Once the modules were detected, the transcripts were separated into sub-modules according to their expression profile.

3.6.3. AgingNet Modules Path Enrichment Analysis

The same procedure for creating enrichment structures developed for the AgingGenes \(described in Section 3.5.4\) was applied to each sub-module, with one change: all “daughter” tracks with Combined Score &lt; 10 were removed. the correlation between age and the median expression profile of the transcripts comprising each enriched pathway was calculated. Correlated pathways \(\|Rho\| &gt; 0.35\) were classified according to their trend \(positive and negative\), and this information was added to the visualization of enrichment structures.

